### PR TITLE
[ready] being jittery or having a chemical speedboost removes spaces, commas, and semicolons from speech

### PR DESCRIPTION
**Pull request by Hatterhat**
idea stolen from vgstation13/30001, see title
image

### CODE REVIEW

**Comment by PastelPrinceDan** [comment]
as a fellow hyperzine (over)user, i approve.

File: code/modules/mob/living/say.dm [hunk 1]
```diff
@@ -110,6 +110,10 @@ var/list/channel_to_radio_key = new
 			S.message = stutter(S.message)
 			verb = pick("stammers","stutters")
 			. = 1
+		if(reagents.has_reagent("hyperzine")) // motor mouthed hyperzine user
```
Comment (Arokha) [inline code/modules/mob/living/say.dm:113]:
```suggestion
		if(reagents?.has_reagent("hyperzine")) // motor mouthed hyperzine user
```
`reagents` is lazy

Also unfortunately this won't work for humans. We have a few different reagent holders, declared on /mob/living/carbon, such as `bloodstr` (the one you should check in this case, I imagine), `ingested`, and `touching`.

You'd have to put this in `carbon/handle_speech_problems()` or, add it to the existing `human/handle_speech_problems()` and read `bloodstr`.

Comment (Hatterhat) [reply code/modules/mob/living/say.dm:113]: somehow, `reagents` on `carbon` actually does point towards `bloodstr`, but `ingested` and `touching` aren't covered, so fair point

i've got a better idea for this, though, now that i've realized that there was a `handle_speech_problems` for humans

Comment (Arokha) [reply code/modules/mob/living/say.dm:113]: Ah, it sets reagents to reference bloodstr in carbon/Initialize(). That's `interesting`. Fair enough.

**Comment by Screemonster** [comment]
I think this'd be funnier as a jitteriness effect so it happens from drinking a shitload of coffee too

**Comment by Hatterhat** [comment]
> I think this'd be funnier as a jitteriness effect so it happens from drinking a shitload of coffee too

you have a fair point

**Review by Arokha — APPROVED** [review]

**Pull request merged**